src/System.Net.HttpListener/src/System/Net/Managed/HttpStreamAsyncResult.cs

```diff
@@ -64,7 +64,7 @@ public void Complete()
                     _handle.Set();
 
                 if (_callback != null)
-                    _callback.BeginInvoke(this, null, null);
+                    _callback(this);
```

Comment:
This is invoking it synchronously rather than asynchronously.  I'm not familiar with this type, but that's a fairly sizable change in behavior.  You could invoke it asynchronously with Task.Run/Task.Factory.StartNew if that's desired.